### ISSUE TITLE
Deleted user appears as "Failed to execute the [useravatar] macro"

### ISSUE THREAD

**Issue by ane-gabriela** [xwikisas/application-xpoll#48]
Steps to reproduce:

1. Log in as Admin
2. Create a poll on a main wiki or subwiki
3. As 2/3 simple users vote
4. Log in as Admin again
5. Delete a user that voted
6. Observe the results in the Poll

Expected results: The name of the user still appears grayed out.

Actual results: "Failed to execute the [useravatar] macro. Cause: [User [XWiki.user02] is not registered in this wiki]. Click on this message for details" 

11.7:
![11 7BeforeDeleteUser](https://user-images.githubusercontent.com/40691112/65269434-d5454080-db21-11e9-890c-d946e013de78.jpg)
![11 7AfterRemoval](https://user-images.githubusercontent.com/40691112/65269433-d4acaa00-db21-11e9-8b59-d802b87c4de4.jpg)

10.11.9:
![10 11 9-DeletedUser](https://user-images.githubusercontent.com/40691112/65269473-e726e380-db21-11e9-992d-b11b6c84a836.jpg)

Environment: Windows 10, XWiki 10.11.9 Standard and 11.7 with MySQL 5.7 and Edge 18

**Comment by CamiAndrei** [comment]
The deleted users are counted in the users tab, the same as their votes. Should they be counted?
![65269433-d4acaa00-db21-11e9-8b59-d802b87c4de4](https://user-images.githubusercontent.com/12369283/66375474-25435480-e9b6-11e9-9640-228d33aec7a3.jpg)

**Comment by oanalavinia** [comment]
IMO the user shouldn't be counted or displayed, as long as he was deleted with a purpose. Considering also the fact that those votes are not deleted, just not displayed (the object is not erased).
If it is to keep counting him, that could also influence the winner of the poll.

There are several general use cases for a poll, like choosing a place to go, or some benefits inside a company, where it would be hard to follow the votes if they contain deleted users.

**Comment by ane-gabriela** [comment]
This ticket was raised as it was an issue for a client. The expectation he had was that once the user was deleted, as in the past, the name was grayed out but the votes would still appear there, as counted.

**Comment by CamiAndrei** [comment]
What about the Polls with the "Finished" status? Should the same behaviour be applied? 
I have identified 3 cases:
1. If the poll's status is "Active", you don't want to take into account the deleted users. As mentioned by Lavinia,  that could also influence the winner of the poll.
2. If the poll has the status "Finished" and the action that should be decided didn't took place, you don't want to display the deleted users.
3. But if the status  is "Finished" and the action already took place ( the poll is keep as history), you want to display the deleted user too.
 The only issue is that between 2 and 3 now is no method to mark them differently (for example "Closed voting" and "Finished" ).

**Comment by oanalavinia** [comment]
If the polls class has no date to set, as you don't always vote for something that will take place in a specific time, then maybe is a valid idea to only consider 2 cases, "Active" with just active users at that moment and "Finished", containing those users that were deleted after the vote was closed.
And maybe the last one is also the client's use case?

**Comment by silviamacovei** [comment]
IMO it makes sense that for the “Finished” state the users are grayed out and the votes still appear and are counted (as Gabi mentioned). Even if the action that was decided on didn’t take place, we would still display the deleted users in gray and count their votes. We don’t have visibility on why the voted action didn’t take place anymore.

For the “Active” state, inactive users shouldn’t be displayed and counted IMO, since the Poll doesn’t concern them anymore. Since the result won’t impact them, they shouldn’t be impacting the result either.

**Comment by lucaa** [comment]
This issue is a little tough to understand, as the description of the issue is talking about a macro error, while the actual implementation seems to have evolved into some hiding of the deleted users _and their votes_ but it's not clear what behaviour was implemented in the end, from all the variations proposed in the comments. It would be great to have a confirmation about which of the discussed behaviours was implemented.
